akka-http-core/src/main/scala/akka/http/scaladsl/model/Multipart.scala

```diff
@@ -43,20 +53,48 @@ sealed trait Multipart {
         .flatMapConcat(ConstantFun.scalaIdentityFunction)
     HttpEntity.Chunked(mediaType withBoundary boundary withCharset charset, chunks)
   }
+
+  /** Java API */
```

Comment:
The Scala side inherits from the Java side, yes.
What would you consider baroque about that? (Apart from having a Java-side in the first place, for everything we do.)
I will add ScalaDoc to all the methods here.